sycl/unittests/helpers/PiMock.hpp

```diff
@@ -130,6 +221,10 @@ class PiMock {
   PiMock(const PiMock &) = delete;
   PiMock &operator=(const PiMock &) = delete;
   ~PiMock() {
+    // Since the plugin relies on the global vars to store function pointers we
+    // need to reset them for the new PiMock plugin instance
+    // TODO: Make function pointers array for each PiMock instance?
```

Comment:
I think we would need that in case gtest ever decides to run tests in parallel. It may be enough to make the arrays `thread_local`. That might have problems when a specific test is multi-threaded though.

Reply:
As an alternative we could:
1) Create a global map `pi_platfrom` to array of function pointers map
2) create a global map from every handle(`void*)` to `pi_platform`
3) update the map(from `2)`) any time we create or remove a handle in the mock functions
4) in the proxy function, for a given handle find a corresponding `pi_platform` using the map from `2)`
5) Using `pi_platform` find an array of function pointers from `1)`

Reply:
I think that might be a good solution to the problem. We could even simplify the map (2) a little, defining a space for which a given platform can use, e.g. the first platform can create handles 0-10000, the next can use 10000-20000, etc. Biggest problem is if someone overwrites the behavior of a function returning handles and return something not in the allocated range/in the map.